src/main/java/org/opensearch/ad/util/ParseUtils.java

```diff
@@ -702,4 +705,24 @@ public static SearchSourceBuilder batchFeatureQuery(
     public static <T> boolean isNullOrEmpty(Collection<T> collection) {
         return collection == null || collection.size() == 0;
     }
+
+    /**
+     * Check if two lists of string equals or not without considering the order.
+     * If the list is null, will consider it equals to empty list.
+     *
+     * @param list1 first list
+     * @param list2 second list
+     * @return true if two list of string equals
+     */
+    public static boolean listEqualsWithoutConsideringOrder(List<String> list1, List<String> list2) {
+        Set<String> set1 = new HashSet<>();
+        Set<String> set2 = new HashSet<>();
+        if (list1 != null) {
+            set1.addAll(list1);
+        }
+        if (list2 != null) {
+            set2.addAll(list2);
+        }
+        return Objects.equals(set1, set2);
+    }
```

Comment:
Seems fine to do this, but what about just sorting with `Collections.sort()` and then using the regular `Object.equals()`, rather than having to maintain this helper function?

Reply:
Here I want to just check if the list equals or not rather than change the lists themselves. `Collections.sort()` is one possible option, but it will change the item order in place. That means after executing this comparing method, the item order will change in original list. Another concern is if the list passed in is immutable, `Connections.sort()` will throw exception.

Reply:
Got it, makes sense.